google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java

```diff
@@ -117,9 +151,11 @@ private static DynamicMessage convertJsonToProtoMessageImpl(
       String jsonLowercaseName = jsonName.toLowerCase();
       String currentScope = jsonScope + "." + jsonName;
       FieldDescriptor field = protoSchema.findFieldByName(jsonLowercaseName);
-      if (field == null) {
+      if (field == null && !ignoreUnknownFields) {
         throw new IllegalArgumentException(
             String.format("JSONObject has fields unknown to BigQuery: %s.", currentScope));
+      } else if (field == null) {
```

Comment:
This check seems redundant.

Reply:
If it is null, it means we cannot proceed on further processing. Otherwise, null ptr will be thrown if we continue.

Reply:
Okay makes sense -- this allows us to skip the unknown field in the middle and proceed onto other fields.